helper.py

```diff
@@ -1,41 +1,186 @@
+=> The following will ONLY run on Python 3.7 to 3.9
+=> Install websockets version 11.0.3 via pip whl (do not use v12)
+=> Install asynio 0.2.0 via pip whl
+"""
 
-import os
+import asyncio
+import aionotify
+from websockets.server import serve
+import websockets
 import time
-
+import base64
+import os
+from io import BytesIO
 from PIL import Image
 
 
-static_path = 'shepherd/static/'
-input_file = static_path + "full_image.jpg"
-output_file = static_path + "image.jpg"
-stat_file = static_path + "imgtime.txt"
+CONNECTIONS = set()
+
+# The following sets up the asynchronous waiting for file change
+picture_watcher = aionotify.Watcher()
+log_watcher = aionotify.Watcher()
+
+img_static_path = "/home/pi/shepherd/shepherd/static/"
+img_input_file = img_static_path + "image.jpg"
+img_output_file = img_static_path + "resized_image.jpg"
+stat_file = img_static_path + "imgtime.txt"
 mod_time = 0
 out_width = 640.0
 out_height = 480.0
 
-while True:
-    time.sleep(0.1)
-    if os.path.exists(input_file):
-        stat_obj = os.stat(input_file)
-        if mod_time != stat_obj.st_mtime:
-            try:
-                img = Image.open(input_file)
-                width_i, height_i = img.size
-                width = width_i * 1.0
-                height = height_i * 1.0
-                scale_x = out_width / width
-                scale_y = out_height / height
-                if scale_x > scale_y:
-                    img = img.resize((int(width*scale_y), int(height*scale_y)), Image.ANTIALIAS)
-                else:
-                    img = img.resize((int(width*scale_x), int(height*scale_x)), Image.ANTIALIAS)
-
-                mod_time = stat_obj.st_mtime
-                img.save(output_file,"JPEG")
-                with open(stat_file, "w") as myfile:
-                    myfile.write(str(mod_time))
-            except IOError:
-                pass
+log_static_path = "/media/RobotUSB/"
+log_input_file = log_static_path + "logs.txt"
+
+picture_watcher.watch(
+    alias="image", path=img_input_file, flags=aionotify.Flags.MODIFY
+)  # sets up watcher
+log_watcher.watch(alias="logs", path=log_input_file, flags=aionotify.Flags.MODIFY)
+
+
+def shrink_image(img):
+    width_i, height_i = img.size
+    width = float(width_i)
+    height = float(height_i)
+    scale_x = out_width / width
+    scale_y = out_height / height
+    if scale_x > scale_y:
+        img = img.resize((int(width * scale_y), int(height * scale_y)), Image.LANCZOS)
+    else:
+        img = img.resize((int(width * scale_x), int(height * scale_x)), Image.LANCZOS)
+    return img
+
+
+def im_2_b64(image):
+    buff = BytesIO()
+    image.save(buff, format="JPEG")
+    img_str = base64.b64encode(buff.getvalue())
+    return img_str
+
+
+async def wait_for_picture_change():
+    loop = asyncio.get_event_loop()
+
+    bypass = False  # so first image is not ignored.
+    while not os.path.exists(img_input_file):
+        await asyncio.sleep(0.5)  # twiddle thumbs :)
```

Comment:
To help other developers (myself included) I would really like this comment to say why we are twiddling our thumbs. I can see that we are.